### PR TITLE
add an end event handler that we can override

### PR DESCRIPTION
**Pull request by joelhooks**
I didn't test this, but the principle is sound 🤡

### CODE REVIEW

File: apps/next-product-starter/src/pages/player.tsx [hunk 1]
```diff
@@ -188,7 +188,13 @@ const VideoCuesList: React.FC<any> = () => {
 const Page = () => {
   return (
     <Layout>
-      <VideoProvider>
+      <VideoProvider
+        actions={{
```
Comment (joelhooks) [inline apps/next-product-starter/src/pages/player.tsx:192]: we can override default actions by sending in an actions object

File: packages/player/src/components/video.tsx [hunk 2]
```diff
@@ -93,6 +93,9 @@ export const Video: React.FC<VideoProps> = ({
       onPlaying={() => {
         videoService.send('DONE_WAITING')
       }}
+      onEnded={() => {
```
Comment (joelhooks) [inline packages/player/src/components/video.tsx:96]: this is the browser api for an ended event. dunno how consistent it is.

https://github.com/joelhooks/cueplayer-react/blob/main/src/components/Video.js#L559

it's what the currently deployed one uses

File: packages/player/src/machines/video-machine.ts [hunk 3]
```diff
@@ -295,7 +295,10 @@ export const videoMachine = createMachine<VideoStateContext, VideoEvent>({
             },
           },
         },
-        ended: {on: {PLAY: 'playing'}},
+        ended: {
+          entry: ['onVideoEnded'],
```
Comment (joelhooks) [inline packages/player/src/machines/video-machine.ts:299]: we weren't doing anything when the playback ended (or even sending the event) but now it should run the action and be prepared to receive events appropriately

**Comment by DrShpongle** [comment]
That works!

**Review by DrShpongle — APPROVED** [review]

**Pull request merged**